packages/middleware/rate-limiter/src/slot.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Slot provider.
+ */
+export interface SlotProvider {
+  /**
+   * Creates an active slot for the given period.
+   */
+  create(args: { fromSecond: number; durationSecond: number; key: string }): Slot
+}
+
+/**
+ * Sliding window slot.
+ */
+export interface Slot {
+  /**
+   * Slot starting time.
+   */
+  readonly fromSecond: number
+  /**
+   * Increases slot internlal counter.
+   */
+  inc(): void
+  /**
+   * Gets slot internal counter.
+   */
+  value(): number
```

Comment:
This is a critical point, need a good documentation because if someone want to implement another support (in-memory, redis, ...) need to understand this

Reply:
No risk of overflowing in my opinion. No risk of performing 9007199254740991 requests without never restarting the server or going to the next slot.